gadgets/trace_exec/program.bpf.c

```diff
@@ -252,4 +253,31 @@ int ig_execve_x(struct syscall_trace_exit *ctx)
 	return 0;
 }
 
+SEC("kprobe/security_bprm_check")
+int BPF_KPROBE(security_bprm_check, struct linux_binprm *bprm)
+{
+	u32 pid = (u32)bpf_get_current_pid_tgid();
+	struct event *event;
+	char *file;
+	struct path f_path;
+
+	if (!paths)
+		return 0;
+
+	event = bpf_map_lookup_elem(&execs, &pid);
+	if (!event)
+		return 0;
+
+	// security_bprm_check is called repeatively following the shebang
+	// Only get the first call.
+	if (event->file[0] != '\0')
```

Comment:
This is for chaining shebangs. With the following example:
* `/tmp/script1.sh`:
```bash
#!/tmp/script2.sh
```
* `/tmp/script2.sh`
```bash
#!/bin/sh
```
The kprobe will be called 3 times:
1. `/tmp/script1.sh`
2. `/tmp/script2.sh`
3. `/bin/sh`

If we only take the last run, we will pick `/bin/sh`, which is the same as exepath, so that would not be useful.

Reply:
Ah I see, thanks!